src/math/Bspline.cpp

```diff
@@ -57,6 +57,12 @@ Bspline::Bspline(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int splineDegree
 
 Eigen::VectorXd Bspline::interpolateAt(double t) const
 {
+  if (math::equals(t, _tsMax)) {
+    t = _tsMax; // set t exactly to _tsMax, to avoid artifacts originating from floating-point arithmetic
```

Comment:
This line might become unreachable when we merge  #1837. Due to that storage always will return the sample when math::equals(t,_tsMax) returns true. I still think that these lines are worth keeping, since they add a safeguard if we change the implementation in storage for when we return the existing sample and when we interpolate.

Reply:
Ok. Then let's merge this PR. Then you can integrate it into #1837 and we can discuss possible redundancies / safeguards over there. I agree with your argument about making the correct functionality of the Bspline class as independent as possible from the implementation of the Storage class.

Reply:
@NiklasKotarsky Added a unit test via 7bae79cf78033ab843d53fdfc39ef11ee0d27ef8. Can you also try this one on your branch? I guess this one should fail, because you are dealing with the issue on the level of the `Storage` and not on the `Bspline`.

Reply:
I just did this on my own, since #1837 was already merged into develop. No bad surprises. I'm just waiting for the test pipeline here to finish and will merge then.

Reply:
Yes, this test fails on my branch.